functions/Find-DbaCommand.ps1

```diff
@@ -0,0 +1,146 @@
+
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/Find-DbaCommand
+
+.EXAMPLE
+Find-DbaCommand "snapshot"
+
+For lazy typers: finds all commands searching the entire help for "snapshot"
+
+.EXAMPLE
+Find-DbaCommand -Pattern "snapshot"
+
+For rigorous typers: finds all commands searching the entire help for "snapshot"
+
+.EXAMPLE
+Find-DbaCommand -Tag copy
+
+Finds all commands tagged with "copy"
+
+.EXAMPLE
+Find-DbaCommand -Author "chrissy"
+
+Finds every command whose author contains our beloved "chrissy"
+
+.EXAMPLE
+Find-DbaCommand -Pattern "snapshot" -Rebuild
+
+Finds all commands searching the entire help for "snapshot", rebuilding the index (good for developers)
+#>
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    Param (
+        [String]$Pattern,
+        [String]$Tag,
+        [String]$Author,
+        [switch]$Rebuild
+    )
+    BEGIN
+    {
+        $tagsRex = ([regex]'(?m)^[\s]{0,15}Tags:(.*)$')
+        $authorRex = ([regex]'(?m)^[\s]{0,15}Author:(.*)$')
+
+        function Get-DbaHelp([String]$commandname) {
+            $thishelp = Get-Help $commandname -Full
+            $thebase = @{}
+            $thebase.CommandName = $commandname
+            $thebase.Name = $thishelp.name
+
+            ## fetch the description
+            $thebase.Description = $thishelp.Description.text
+
+            ## fetch examples
+            $thebase.Examples = $thishelp.Examples | Out-String -Width 120
+
+            ## fetch help link
+            $thebase.Links = ($thishelp.relatedLinks).navigationLink.uri
+
+            ## fetch the synopsis
+            $thebase.Synopsis = $thishelp.Synopsis
+
+            ## store notes
+            $as = $thishelp.alertSet | Out-String -Width 120
+
+            ## fetch the tags
+            $tags = $tagsrex.Match($as).Groups[1].Value
+            if($tags) {
+                $thebase.Tags = $tags.Trim().Split(',')
+            }
+            ## fetch the author
+            $author = $authorRex.Match($as).Groups[1].Value
+            if($author) {
+                $thebase.Author = $author.Trim()
+            }
+
+            [pscustomobject]$thebase
+        }
+
+        function Get-DbaIdxPath() {
+            $docs = [Environment]::GetFolderPath("mydocuments")
+            $idxpath = Join-Path $docs "dbatools-index.json"
+            $idxpath
+        }
+
+        function Get-DbaIndex() {
+            $dbamodule = Get-Module dbatools
+            $allcommands = $dbamodule.ExportedCommands
+            $helpcoll = New-Object System.Collections.Generic.List[System.Object]
+            foreach($c in $allcommands.GetEnumerator()) {
+                $x = Get-DbaHelp $c.Key
+                $helpcoll.Add($x)
+            }
+            $dest = Get-DbaIdxPath
+            if ($Pscmdlet.ShouldProcess($dest, "Recreating index"))
```

Comment:
sure, TBF I added it at the last moment just because out-file without a shouldprocess gets tagged by scriptanalyzer.